packages/fxa-auth-server/lib/payments/iap/apple-app-store/apple-iap.ts

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Firestore } from '@google-cloud/firestore';
+import { AppStoreServerAPI, Environment } from 'app-store-server-api';
+import { Container } from 'typedi';
+
+import { AppConfig, AuthFirestore, AuthLogger } from '../../../types';
+import { PurchaseManager } from './purchase-manager';
+
+export class AppleIAP {
+  private firestore: Firestore;
+  private log: AuthLogger;
+  private prefix: string;
+
+  public purchaseManager: PurchaseManager;
+  constructor() {
+    this.log = Container.get(AuthLogger);
+    const {
+      authFirestore,
+      subscriptions: { appStore },
+    } = Container.get(AppConfig);
+
+    // Initialize App Store Server API client per bundle ID
+    const environment = appStore.sandbox
+      ? Environment.Sandbox
+      : Environment.Production;
+    const appStoreServerApiClients = {};
+    for (const [bundleIdWithUnderscores, credentials] of Object.entries(
+      appStore.credentials
+    )) {
+      // Cannot use an actual bundleId (e.g. 'org.mozilla.ios.FirefoxVPN') as the key
+      // due to https://github.com/mozilla/node-convict/issues/250
+      const bundleId = bundleIdWithUnderscores.replace('_', '.');
+      const { serverApiKey, serverApiKeyId, issuerId } = credentials;
+      appStoreServerApiClients[bundleId] = new AppStoreServerAPI(
+        serverApiKey,
+        serverApiKeyId,
+        issuerId,
+        bundleId,
+        environment
+      );
```

Comment:
What is in charge of managing which bundleId to use for which API request? I thought this was going to be folded into a class that handles it automatically.

Reply:
As we discussed, I created a new class to handle this, `AppStoreHelper`.